mappings/net/minecraft/client/render/item/ItemRenderer.mapping

```diff
@@ -54,6 +54,19 @@ CLASS net/minecraft/class_918 net/minecraft/client/render/item/ItemRenderer
 		ARG 4 overlay
 		ARG 5 matrices
 		ARG 6 vertices
+	METHOD method_27951 renderInGui (Lnet/minecraft/class_1309;Lnet/minecraft/class_1799;II)V
+		COMMENT Renders an item in a GUI with an attached entity.
+		COMMENT
+		COMMENT <p>The entity is used to calculate model overrides for the item.
+		ARG 1 entity
+		ARG 2 stack
+		ARG 3 x
+		ARG 4 y
+	METHOD method_27953 renderInGuiWithoutEntity (Lnet/minecraft/class_1799;II)V
```

Comment:
would recommend this to be renderGuiItem and the player one with renderGuiItemWithPlayer

Reply:
or I meant renderItemInGuiWithPlayer, depends which one you prefer.

imo this is because calling with player is a "special case" for it, and calling without an entity is not

Reply:
I feel like the player one is the default one that should be used in GUIs, since for example `HandledScreen` does that.

Reply:
HandledScreen is player specific then I guess, I don't feel like we should attach the player without the withPlayer thing at least, the player can be null, which makes this the same as no entity. So let the modder know by adding the withPlayer thing

Reply:
I can run a usage search in Enigma and see which one Mojang considers the "default" one (by number of usages).

Reply:
With the player:
![search results](https://user-images.githubusercontent.com/6596629/82597547-cd0e3b00-9bb1-11ea-9f8a-bae9788df948.png)

Without an entity:
![search results](https://user-images.githubusercontent.com/6596629/82597583-da2b2a00-9bb1-11ea-91b8-2894385d2130.png)

With a manually specified entity (which is the client player in all cases):
![search results](https://user-images.githubusercontent.com/6596629/82597638-f7f88f00-9bb1-11ea-8e37-d80c225a6628.png)

Reply:
Does the renderer apply model overrides with the player given, or what is the player used for?

Reply:
It is for the overrides predicate, for example, the one for `lefthanded`, I would say that the player one is mainly for handled screens because it is to detect properties for the player, so it would be essential to add `WithPlayer` at the end to inform the modder that it will use the player.

For things like the advancement tabs, they are not player dependent, you don't want to see the icon change due to the player doing some weird things.

Reply:
then `WithOverrides` suffix is better than `WithPlayer` suffix.

Reply:
Is this distinction necessary? Can they both not just be `renderInGui`? The arguments and documentation can describe other behaviours.

Reply:
this is, as this method does not allow item model's overrides to be checked or applied.

Reply:
@ChloeDawn There are conflicting signatures (`ItemStack, int, int`) with two of them (the player one and the no-entity one), so they have to have different names.